kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowViewStub.kt

```diff
@@ -0,0 +1,62 @@
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.View
+import android.view.ViewGroup
+
+/**
+ * A placeholder [View] that can replace itself with ones driven by workflow renderings,
+ * similar to [android.view.ViewStub].
+ */
+class WorkflowViewStub(
+  context: Context,
+  attributeSet: AttributeSet? = null
+) : View(context, attributeSet) {
+  init {
+    setWillNotDraw(true)
+  }
+
+  /**
+   * By starting as our own delegate, we make the parent checking code in [update] simpler.
+   */
+  private var _actual: View = this
+
+  /**
+   * On-demand access to the delegate established by the last call to [update],
+   * or null if none has been set yet.
+   */
+  val actual: View? get() = _actual.takeIf { it !is WorkflowViewStub }
+
+  /**
+   * Replaces this view with one that can display [rendering]. If it
+   * has already been replaced, updates that replacement if it [canShowRendering].
+   * If it can't, a new replacement is put in place.
```

Comment:
"Place" is starting to sound funny after reading this 🤪

Reply:
```
   * Replaces this view with one that can display [rendering]. If the receiver
   * has already been replaced, updates the replacement if it [canShowRendering].
   * If the current replacement can't handle [rendering], a new view is put in place.
```